src/main/java/com/endava/internship/collections/StudentMap.java

```diff
@@ -1,72 +1,396 @@
 package com.endava.internship.collections;
 
-import java.util.Collection;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Consumer;
 
 public class StudentMap implements Map<Student, Integer> {
+
+    private final int DEFAULT_CAPACITY = 1 << 4;
+    private final int MAXIMUM_CAPACITY = 1 << 30;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    final class Node {
+        final int hash;
+        final Student key;
+        Integer value;
+        Node next;
+
+        Node(int hash, Student key, Integer value, Node next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+        public final Student getKey() {
```

Comment:
no need the final in these methods. no one will ever capable to override your class since you used the final keyword on the class lvl

Reply:
Fixed